src/services/role/services/rolesService.js

```diff
@@ -0,0 +1,89 @@
+const { RoleModel } = require('../model');
+
+// do not proteced this route with authentication
+const RoleServiceHooks = {
+	before: {
+		all: [],
+		find: [],
+		get: [],
+	},
+};
+
+const unique = (base, merge) => [...new Set([...base, ...merge])];
+
+const dissolveInheritPermission = (roles, role) => {
+	if (Array.isArray(role.roles) && role.roles[0]) {
+		const inheritRoleId = role.roles[0].toString();
+		const inheritRole = roles.find((r) => r._id.toString() === inheritRoleId);
+		const { permissions } = dissolveInheritPermission(roles, inheritRole);
+		role.permissions = unique(role.permissions, permissions);
+	}
+	return role;
+};
+
+const removeKeys = (keys = []) => (role) => {
+	keys.forEach((key) => {
+		delete role[key];
+	});
+	return role;
+};
+
+const paginate = (result, query = {}) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected block statement surrounding arrow body; parenthesize the returned value and move it immediately after the `=>`.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5229889)